### ISSUE TITLE
Freeze when closing or switching from an app that tears

### ISSUE THREAD

**Issue by FakeMichau** [hyprwm/Hyprland#3463]
### Hyprland Version

Hyprland, built from branch main at commit a2a29a60e5301df2cadf58c1bb18495d02710547 dirty (tearing: cleanup vars and avoid rendering before drm is ready).

### Bug or Regression?

Bug

### Description

The desktop freezes when switching from a fullscreen app that tears to one that doesn't. 
9 out of 10 times it will freeze instead of switching. Closing the app has the same result.
Is caused by https://github.com/hyprwm/Hyprland/commit/a2a29a60e5301df2cadf58c1bb18495d02710547#diff-b12458cc1c330ab36cf13b6652d2babf94b8dc91408d83a8d3ec48ef238ca156R156-R157

### How to reproduce

Have tearing enabled on a fullscreen app (vrrtest in my case) on one workspace and attempt to switch to another workspace. Usually will freeze the desktop but sometimes it will be successful

**Comment by vaxerski** [comment]
the "don't leave fullscreen on new window" is on purpose

**Comment by FakeMichau** [comment]
> the "don't leave fullscreen on new window" is on purpose
That's not what I'm seeing here. 

First of all this only happens if I visually can see tearing, VRR has to be disabled. VRRTest is just the easiest app to verify that.

When VRRTest is running and tearing on fullscreen on workspace 1, trying to switch to workspace 2 (that has nothing open) will more often than not stop the rendering of my desktop and show me a stop frame of VRRTest. Mouse is also frozen.
While in that state, switching back to the workspace 1 will resume the rendering, VRRTest is still running, I can move my mouse.
Doing it over and over I'm sometimes able to successfully switch to workspace 1 and see my wallpaper

Just closing the VRRTest will also cause this and I haven't found a way of recovering from that requiring me to kill Hyprland.

I've used ``windowrulev2 = immediate,class:(love)`` to force tearing in VRRTest

I just found that out - sometimes it just work as it should. But restarting Hyprland will cause the bug to appear again.
